Support curl for MacOS (or all?)

Building this on MacOS can break as it uses `wget` in the Makefile which isn't usually default on MacOS system:

```
wget -O /Users/jonjon/repos/smartrent_hub/smartrent_hub_fw/deps/nerves_time_zones/tzdata2021a.tar.gz https://data.iana.org/time-zones/releases/tzdata2021a.tar.gz
make[1]: wget: No such file or directory
make[1]: *** [/Users/jonjon/repos/smartrent_hub/smartrent_hub_fw/deps/nerves_time_zones/tzdata2021a.tar.gz] Error 1
make: *** [tzdata] Error 2
```

Its a simple `brew install wget` fix I believe, but documenting here in case we want to change

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -16,7 +16,7 @@ install_system_deps: &install_system_deps
   run:
     name: Install system dependencies
     command: |
-        apk add build-base
+        apk add build-base curl
 
 jobs:
   build_elixir_1_12_otp_24:

--- a/tzdata.mk
+++ b/tzdata.mk
@@ -65,7 +65,7 @@ $(PREFIX)/zoneinfo: $(BUILD)/zic $(BUILD)/tzdata/.extracted Makefile
 	cd $(BUILD)/tzdata && $(BUILD)/zic -d $@ $(ZIC_OPTIONS) $(TDATA)
 
 $(TZDATA_ARCHIVE_PATH):
-	wget -O $(TZDATA_ARCHIVE_PATH) $(TZDATA_URL)
+	curl -L $(TZDATA_URL) > $@
 
 $(BUILD)/tzdata/.extracted: $(TZDATA_ARCHIVE_PATH)
 	mkdir -p $(BUILD)/tzdata

